### PR TITLE
feat: React-Native-MacOS Support

### PR DESCRIPTION
**Pull request by bralax**
# Summary
I am working on a project using [react-native-macos](https://github.com/microsoft/react-native-macos) and needed access to a picker, so I decided to implement this library for macOS using the NSPopUpButton.

## Test Plan

To test that it worked, I built a [small sample application](https://github.com/bralax/react-native-picker-macos-example) that uses the library.

Here is some screenshots of it in use.
Screen Shot 2020-07-30 at 12 01 33 AM
Screen Shot 2020-07-30 at 12 01 43 AM
Screen Shot 2020-07-30 at 12 02 10 AM

### What's required for testing (prerequisites)?
A macOS computer running >= 10.14
### What are the steps to reproduce (after prerequisites)?
1. Clone the sample application -> won't work in a normal repo currently as pod spec does not work as it will always download from this repo
2. Run `yarn`
3. Replace the picker node module with this code
4. Pod install
5. Run the code with `npx react-native run-macos`
## Compatibility

| OS      | Implemented |
| -------  | :---------: |
| iOS        |    ❌     |
| Android |    ❌     |
| MacOS |    ✅     |

## Checklist

- [X] I have tested this on a device and a simulator
- [X] I added the documentation in `README.md`
- [X] I mentioned this change in `CHANGELOG.md` -> Does not exist?
- [ ] I updated the typed files (TS and Flow)
- [X] I added a sample use of the API in the example project (`example/App.js`) -> Does not currently work as the pod spec relies on this GitHub and the code for macOS is currently not in the pod which is why I made the sample project with a modified pod spec.

### CODE REVIEW

File: macos/RNCPicker.m [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+  /*if ((self = [super initWithFrame:frame])) {
+    _color = [UIColor blackColor];
+    _font = [UIFont systemFontOfSize:21]; // TODO: selected title default should be 23.5
+    _selectedIndex = NSNotFound;
+    _textAlign = NSTextAlignmentCenter;
+    self.delegate = self;
+    self.dataSource = self;
+    [self selectRow:0 inComponent:0 animated:YES]; // Workaround for missing selection indicator lines (see https://stackoverflow.com/questions/39564660/uipickerview-selection-indicator-not-visible-in-ios10)
+  }
+  return self;*/
```
Comment (Naturalclar) [inline macos/RNCPicker.m:46]: if there a reason for this comment to be here?
if so, could you add a comment in the code :+1:

File: macos/RNCPicker.m [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+{
+  _selectedIndex = row;
+  if (_onChange && _items.count > (NSUInteger)row) {
+    _onChange(@{
+      @"newIndex": @(row),
+      @"newValue": RCTNullIfNil(_items[row][@"value"]),
+    });
+  }
+}
+*/
```
Comment (Naturalclar) [inline macos/RNCPicker.m:151]: same as comment above

File: macos/RNCPicker.m [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+//RCT_NOT_IMPLEMENTED(- (instancetype)initWithCoder:(NSCoder *)aDecoder)
+
+- (void)setItems:(NSArray<NSDictionary *> *)items
+{
+    _items = [items copy];
+    /*NSDictionary *attributes = [NSDictionary
+    dictionaryWithObjectsAndKeys:
+    color, NSForegroundColorAttributeName,
+    font,
+    NSFontAttributeName, textAlign, NSParagraphStyleAttributeName, nil];*/
```
Comment (Naturalclar) [inline macos/RNCPicker.m:59]: same as comment above

**Comment by bralax** [comment]
Hi @Naturalclar That was my fault for leaving those in and I have no removed the comments. I was using the IOS implementation as a reference so a lot of that is that as I went I commented out code that did not apply to macOS (for example the macOS equivalent does not use a delegate or datasource) and forgot to delete afterwards.

**Review by Naturalclar — APPROVED** [review]
@bralax thank you so much!

**Pull request merged**